 # Machine Learning Assignment Week 3 Task 2 

###### importing required libraries and files 

In [2]:
import pandas as pd
df = pd.read_csv('task1.csv')
df.head()

PostCode      Borough                          Neighborhood
0      M1B  Scarborough                         Rouge,Malvern
1      M1C  Scarborough  Highland Creek,Rouge Hill,Port Union
2      M1E  Scarborough       Guildwood,Morningside,West Hill
3      M1G  Scarborough                                Woburn
4      M1H  Scarborough                             Cedarbrae

###### Installing the geocoder library 

In [3]:
!pip install geocoder
import geocoder

     |████████████████████████████████| 102kB 3.2MB/s ta 0:00:011
     |████████████████████████████████| 829kB 12.2MB/s eta 0:00:01
  Stored in directory: /home/jovyan/.cache/pip/wheels/74/bc/50/ae030267a192919b289f84661cfeb5573cf383841c543e8696
Successfully built future


In [4]:
def get_latlng(postal_code):
    # initialize your variable to None
    lat_lng_coords = None
    # loop until you get the coordinates
    while(lat_lng_coords is None):
        g = geocoder.arcgis('{}, Toronto, Ontario'.format(postal_code))
        lat_lng_coords = g.latlng
    return lat_lng_coords
    
get_latlng('M4G')

[43.70949500000006, -79.36398897099997]

In [8]:
post_codes = df['PostCode']    
coords = [ get_latlng(post_code) for post_code in post_codes.tolist() ]

In [9]:
df_coords = pd.DataFrame(coords, columns=['Latitude', 'Longitude'])
df['Latitude'] = df_coords['Latitude']
df['Longitude'] = df_coords['Longitude']

In [10]:
df.head(12)

PostCode      Borough                                       Neighborhood  \
0       M1B  Scarborough                                      Rouge,Malvern   
1       M1C  Scarborough               Highland Creek,Rouge Hill,Port Union   
2       M1E  Scarborough                    Guildwood,Morningside,West Hill   
3       M1G  Scarborough                                             Woburn   
4       M1H  Scarborough                                          Cedarbrae   
5       M1J  Scarborough                                Scarborough Village   
6       M1K  Scarborough          East Birchmount Park,Ionview,Kennedy Park   
7       M1L  Scarborough                      Clairlea,Golden Mile,Oakridge   
8       M1M  Scarborough      Cliffcrest,Cliffside,Scarborough Village West   
9       M1N  Scarborough                         Birch Cliff,Cliffside West   
10      M1P  Scarborough  Dorset Park,Scarborough Town Centre,Wexford He...   
11      M1R  Scarborough                                   Maryvale,Wexford   

     Latitude  Longitude  
0   43.811525 -79.195517  
1   43.785665 -79.158725  
2   43.765815 -79.175193  
3   43.768369 -79.217590  
4   43.769688 -79.239440  
5   43.743125 -79.231750  
6   43.726276 -79.263625  
7   43.713054 -79.285055  
8   43.724235 -79.227925  
9   43.696770 -79.259967  
10  43.759975 -79.268974  
11  43.750710 -79.300560

In [11]:
df.shape

(103, 5)

##### This suffices the assignment requirement of producing a dataframe containing 103 rows and 5 columns including logitude and lattitude coordinates. The first 12 rows have been displayed. 

##### Saving for future

In [12]:
df.to_csv('task2.csv', index=False)